Sources/Mustache/Template+Parser.swift

```diff
@@ -306,16 +306,44 @@ extension MustacheTemplate {
         var nameParser = Parser(String(text))
         let string = nameParser.read(while: self.sectionNameCharsWithoutBrackets)
         if nameParser.reachedEnd() {
-            return (text, nil)
+            return (text, [])
         } else {
+            var transforms: [Substring] = [string]
+            var parameterName: Substring?
+
+            func parseTransforms() throws {
+                if let previous = parameterName {
+                    transforms.append(previous)
```

Comment:
Any reason you don't add the transform at the point you hit a "(" . ie line 332

Reply:
That ... won't be any better.
We need to capture the current `parameterName` and append it to the `transforms`, then override the `parameterName`.

Reply:
It just seems a little clearer to me. You are adding a transform at the point you've recognised it is a transform. Not somewhere later. Also removes a branch although that isn't overly important as this code isn't called at a point you need performance.

Reply:
I can try if you insist, but it'll make the code weirder.
I understand the better clarity, but the change didn't seem that simple when i wanted to actually do it. It's not as simple as just moving that line of code.

Reply:
Actually ... it looks pretty weird I'm not sure how to properly do it...
Can you suggest what you think i should change the code to, without breaking the functionality?

Reply:
To me it looks like the code is in the correct place.
Maybe the recursive functions isn't clear enough.

```swift
                if let previous = parameterName {
                    transforms.append(previous)
                }
                parameterName = nameParser.read(while: self.sectionNameCharsWithoutBrackets)
```
We have a new name to read, and only at the point immediately be fore that, we are recognizing the previously parameter-name as a transform and adding it to the array.

Reply:
Did you mean the placement of the function? Now I see ...🫣

Reply:
```swift
func parseTransforms() throws -> Substring {
    // read the next parameter-name candidate
    let parameterName = nameParser.read(while: self.sectionNameCharsWithoutBrackets)
    switch nameParser.current() {
    case ")":
        // Transforms are ending
        nameParser.unsafeAdvance()
        // We need to have a `)` for each transform that we've parsed
        guard nameParser.read(while: ")") + 1 == transforms.count else {
            throw Error.unfinishedName
        }
        return parameterName
    case "(":
        // Parse the next transform
        nameParser.unsafeAdvance()
        transforms.append(parameterName)
        return try parseTransforms()
    default:
        throw Error.unfinishedName
    }
}
let parameterName = try parseTransforms()
```

Reply:
Alright, did that with some adjustments.